File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        set<int> set;
        for(int i = 0; i < nums.size(); i++){
            if(set.find(nums[i]) != set.end()){
                return true;
                
            }
            else{
                set.insert(nums[i]);
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:

        res = []

        def dfs(i, curr, total):
            if total == target: # find target
                res.append(curr.copy())
                return 
            if i >= len(nums) or total >target: # no solution down this path
                return 

            curr.append(nums[i]) # include same candidate
            dfs(i, curr, total + nums[i])

            curr.pop() # don't include new candidate
            dfs(i + 1, curr, total)

        dfs(0, [], 0)
        return res


        